src/Bicep.Core/TypeSystem/DiscriminatedObjectType.cs

```diff
@@ -11,11 +11,11 @@ public class DiscriminatedObjectType : TypeSymbol
         public DiscriminatedObjectType(string name, TypeSymbolValidationFlags validationFlags, string discriminatorKey, IEnumerable<ITypeReference> unionMembers)
             : base(name)
         {
-            var unionMembersByKey = new Dictionary<string, ITypeReference>();
+            var unionMembersByKey = new Dictionary<string, NamedObjectType>();
```

Comment:
>NamedObjectType [](start = 59, length = 15)

Wouldn't this block nested discriminators?

Reply:
Yes, but if you take a look at line 18 - you'll see we're already blocking it. I'd suggest opening an issue if you think it's something we should support.